### PR TITLE
[SHELL32][SHELL32_APITEST][SDK] Implement GUIDFromStringA

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
Implementing missing features...
JIRA issue: [CORE-19278](https://jira.reactos.org/browse/CORE-19278)

## Proposed changes

- Add `GUIDFromString` testcase.
- Implement `GUIDFromStringA` function.
- Add NULL check to `GUIDFromStringW`.
- Add `GUIDFromStringA` prototype.

## TODO

- [ ] Do big tests.

## Comparison

WinXP:
WinXP
Successful.

Win2k3:
Win2k3
Successful.

Win10 x64:
Win10
Successful.

ReactOS (BEFORE):
before
Some failures.

ReactOS (AFTER):
after
Successful.

### CODE REVIEW

**Comment by katahiromz** [comment]
Test KVM (BEFORE): 938a39a https://reactos.org/testman/compare.php?ids=94604
Test KVM (AFTER): cfa6b47

Test VBox (BEFORE): 938a39a https://reactos.org/testman/compare.php?ids=94603
Test VBox (AFTER): cfa6b47

File: modules/rostests/apitests/shell32/GUIDFromString.cpp [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+//            0x00, 0x00, 0x00, 0x46);
+//DEFINE_GUID(IID_IShellLinkW_Invalid, 0x000214F9, 0x0000, 0x0000, 0xC0, 0x00, 0x00, 0x00,
+//            0x00, 0x00, 0x00, 0xED);
+
+static void TEST_GUIDFromStringA(void)
+{
+    GUID guid;
+
+    guid = invalid_guid;
+    ok_int(GUIDFromStringA("", &guid), FALSE);
```
Comment (learn-more) [inline modules/rostests/apitests/shell32/GUIDFromString.cpp:25]: should it crash when receiving a `nullptr` for either of the arguments?

Comment (katahiromz) [reply modules/rostests/apitests/shell32/GUIDFromString.cpp:44]: Fixed.

**Comment by katahiromz** [comment]
Test KVM (BEFORE): 938a39a
Test KVM (AFTER): 9fad795

Test VBox (BEFORE): 938a39a
Test VBox (AFTER): 9fad795

**Pull request merged**